python/pyarrow/types.pxi

```diff
@@ -1591,6 +1591,100 @@ cdef class ExtensionType(BaseExtensionType):
         return ExtensionScalar
 
 
+cdef class VariableShapeTensorType(BaseExtensionType):
+    """
+    Concrete class for variable shape tensor extension type.
+
+    Examples
+    --------
+    Create an instance of variable shape tensor extension type:
+
+    >>> import pyarrow as pa
+    >>> pa.variable_shape_tensor(pa.int32(), 2)
+    VariableShapeTensorType(extension<arrow.variable_shape_tensor>)
```

Comment:
Can you add a repr similar as we have for FixedShapeTensorType (that includes the parametrization)

Reply:
Good point! Added.